libpod/runtime.go

```diff
@@ -948,6 +950,24 @@ func makeRuntime(runtime *Runtime) (err error) {
 	// further
 	runtime.valid = true
 
+	if runtime.doMigrate {
+		if os.Geteuid() != 0 {
+			aliveLock.Unlock()
```

Comment:
We want to be holding the alive lock here, to ensure that other Podman instances don't start while we're still migrating containers

Reply:
If the BecomeRoot call prevents us from grabbing it here, migrate() itself should take the lock and hold it

Reply:
once we re-exec, the rootless instance will run with euid==0 and it will hold the lock.  This is to prevent a deadlock with the re-execed process